java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -522,7 +540,137 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
+    }
+  }
+
+  /**
+   * Concatenate columns of strings together using a separator specified for each row
+   * and returns the result as a string column. If the row separator for a given row is null,
+   * output column for that row is null unless separator_narep is provided.
+   * The separator is applied between two output row values if the separate_nulls
+   * is `YES` or only between valid rows if separate_nulls is `NO`.
+   * @param columns array of columns containing strings, must be more than 1 columns
+   * @param sep_col strings column that provides the separator for a given row
+   * @param separator_narep String that should be used in place of a null separator for a given
+   *                        row. 
+   * @param col_narep string that should be used in place of any null strings
+   *                  found in any column.
+   * @param separate_nulls if true, then the separator is included for null rows if
+   *                       `col_narep` is valid.
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenate(ColumnView[] columns,
+      ColumnView sep_col, Scalar separator_narep, Scalar col_narep, boolean separate_nulls) {
+    assert columns.length >= 1 : ".stringConcatenate() operation requires at least 1 column";
+    assert separator_narep != null : "separator narep scalar provided may not be null";
+    assert col_narep != null : "column narep scalar provided may not be null";
+    assert separator_narep.getType().equals(DType.STRING) : "separator naprep scalar must be a string scalar";
+    assert col_narep.getType().equals(DType.STRING) : "column narep scalar must be a string scalar";
+
+    long[] column_views = new long[columns.length];
+    for(int i = 0; i < columns.length; i++) {
+      assert columns[i] != null : "Column vectors passed may not be null";
+      column_views[i] = columns[i].getNativeView();
+    }
+
+    return new ColumnVector(stringConcatenationSepCol(column_views, sep_col.getNativeView(),
+      separator_narep.getScalarHandle(), col_narep.getScalarHandle(), separate_nulls));
+  }
+
+  /**
+   * Given a lists column of strings (each row is a list of strings), concatenates the strings
+   * within each row and returns a single strings column result. Each new string is created by
+   * concatenating the strings from the same row (same list element) delimited by the separator
+   * provided. This version of the function relaces nulls with empty string and returns null
+   * for empty list.
+   * @param list_column column containing lists of strings to concatenate.
+   * @param sep_col strings column that provides separators for concatenation.
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenateListElements(ColumnView list_column,
```

Comment:
Curious that this is a static function on a single column view instead of a member function for that column view.  Comment applies to the other forms where we just need a main list column to drive the concatenate,

Reply:
yes good point, I'll move them